bot/exts/utilities/colour.py

```diff
@@ -0,0 +1,256 @@
+            raise commands.BadArgument(
+                message=f"RGB values can only be from 0 to 255. User input was: `{red, green, blue}`."
+            )
+        rgb_tuple = (red, green, blue)
+        await self.send_colour_response(ctx, rgb_tuple)
+
+    @colour.command()
+    async def hsv(self, ctx: commands.Context, hue: int, saturation: int, value: int) -> None:
+        """Command to create an embed from an HSV input."""
+        if (hue not in range(361)) or any(c not in range(101) for c in (saturation, value)):
+            raise commands.BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Value can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, value}`."
+            )
+        hsv_tuple = ImageColor.getrgb(f"hsv({hue}, {saturation}%, {value}%)")
+        await self.send_colour_response(ctx, hsv_tuple)
+
+    @colour.command()
+    async def hsl(self, ctx: commands.Context, hue: int, saturation: int, lightness: int) -> None:
+        """Command to create an embed from an HSL input."""
+        if (hue not in range(361)) or any(c not in range(101) for c in (saturation, lightness)):
+            raise commands.BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Lightness can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, lightness}`."
+            )
+        hsl_tuple = ImageColor.getrgb(f"hsl({hue}, {saturation}%, {lightness}%)")
+        await self.send_colour_response(ctx, hsl_tuple)
+
+    @colour.command()
+    async def cmyk(self, ctx: commands.Context, cyan: int, magenta: int, yellow: int, key: int) -> None:
+        """Command to create an embed from a CMYK input."""
+        if any(c not in range(101) for c in (cyan, magenta, yellow, key)):
+            raise commands.BadArgument(
+                message=f"CMYK values can only be from 0 to 100. User input was: `{cyan, magenta, yellow, key}`."
+            )
+        r = round(255 * (1 - (cyan / 100)) * (1 - (key / 100)))
+        g = round(255 * (1 - (magenta / 100)) * (1 - (key / 100)))
+        b = round(255 * (1 - (yellow / 100)) * (1 - (key / 100)))
+        await self.send_colour_response(ctx, (r, g, b))
+
+    @colour.command()
+    async def hex(self, ctx: commands.Context, hex_code: str) -> None:
+        """Command to create an embed from a HEX input."""
+        if hex_code[0] != "#":
+            hex_code = f"#{hex_code}"
+
+        if len(hex_code) not in (4, 5, 7, 9) or any(_ not in string.hexdigits for _ in hex_code[1:]):
+            hex_error_embed = discord.Embed(
+                title="The input hex code is not valid.",
+                message=f"Cannot convert `{hex_code}` to a recognizable Hex format. "
+                "Hex values must be hexadecimal and take the form *#RRGGBB* or *#RGB*.",
+                colour=discord.Colour.dark_red()
+            )
+            await ctx.send(hex_error_embed)
```

Comment:
```suggestion
            hex_error_embed = discord.Embed(
                title="The input hex code is not valid.",
                description=f"Cannot convert `{hex_code}` to a recognizable Hex format. "
                "Hex values must be hexadecimal and take the form *#RRGGBB* or *#RGB*.",
                colour=discord.Colour.dark_red()
            )
            await ctx.send(embed=hex_error_embed)
```
change on line 151 should also be made on 250. Will the other color modes have similar error embeds as well?

Reply:
It should be raised by the BadArguments, with the details of the accepted ranges for the other commands. 

This is what I have locally for the `match_colour_name`:
```py
    async def match_colour_name(self, ctx: commands.Context, input_colour_name: str) -> Union[str, None]:
        """Convert a colour name to HEX code."""
        try:
            match, certainty, _ = rapidfuzz.process.extractOne(
                query=input_colour_name,
                choices=self.colour_mapping.keys(),
                score_cutoff=80
            )
        except (ValueError, TypeError):
            name_error_embed = discord.Embed(
                title="No colour match found.",
                description=f"No color found for: `{input_colour_name}`",
                colour=discord.Color.dark_red()
            )
            await ctx.send(name_error_embed)
        return f"#{self.colour_mapping[match]}"
```

Reply:
Added the fixes for `description` and `embed=`, seems like it is working now.